### PR TITLE
Add GPT-4o mini support

### PR DESCRIPTION
**Pull request by PeterDaveHello**
GPT-4o mini 🎉 

Reference:
- https://openai.com/api/pricing/
- https://platform.openai.com/docs/models/gpt-4o-mini
- https://openai.com/index/gpt-4o-mini-advancing-cost-efficient-intelligence/

## Summary by CodeRabbit

- **New Features**
	- Introduced the `gpt-4o-mini` model, offering enhanced capabilities, affordability, and speed.
	- Updated model selection component to include and allow selection of the `gpt-4o-mini` model.

- **Improvements**
	- Enhanced model descriptions and cost information for better user understanding.
	- Updated model mappings to reflect new options, including token limits for the `gpt-4o-mini`.

### CODE REVIEW

**Review by Christopher-Hayes — APPROVED** [review]
:tada:

**Comment by PeterDaveHello** [comment]
@coderabbitai

**Comment by PeterDaveHello** [comment]
@coderabbitai full

File: src/renderer/components/ModelSelect.tsx [hunk 1]
```diff
@@ -798,6 +799,40 @@ export default function ModelSelect({
...
+                        {MODEL_TOKEN_LIMITS.get(gpt4o.id)?.max && (
+                          <>
+                            , Completion:{" "}
+                            <code>{MODEL_TOKEN_LIMITS.get(gpt4o.id)?.max}</code>
+                          </>
+                        )}
+                      </p>
+                    </button>
+                  );
+                })()}
```
[the remark this answers is not in the document]
Comment (PeterDaveHello) [reply src/renderer/components/ModelSelect.tsx]: How nice!

**Comment by Christopher-Hayes** [comment]
LGTM

**Pull request merged**